Source/Csla/Configuration/Fluent/CslaDataPortalConfiguration.cs

```diff
@@ -85,10 +125,53 @@ public CslaDataPortalConfiguration AutoCloneOnUpdate(bool value)
     /// the client as part of the DataPortalException.
     /// </summary>
     /// <param name="value">Value (default is false)</param>
-    public CslaDataPortalConfiguration DataPortalReturnObjectOnException(bool value)
+    public bool DataPortalReturnObjectOnException
     {
-      ConfigurationManager.AppSettings["CslaDataPortalReturnObjectOnException"] = value.ToString();
-      return this;
+      get
+      {
+        bool result = false;
+        string setting = ConfigurationManager.AppSettings["CslaDataPortalReturnObjectOnException"];
+        if (!string.IsNullOrEmpty(setting))
+          result = bool.Parse(setting);
+        return result;
+      }
+      set
+      {
+        ConfigurationManager.AppSettings["CslaDataPortalReturnObjectOnException"] = value.ToString();
+      }
     }
+
+    /// <summary>
+    /// Gets or sets a value containing the type of the
+    /// IDataPortalAuthorizer to be used by the data portal.
+    /// An instance of this type is created using dependency
+    /// injection.
+    /// </summary>
+    public Type AuthorizerProviderType { get; set; } = typeof(ActiveAuthorizer);
+
+    /// <summary>
+    /// Gets a list of the IInterceptDataPortal instances
+    /// that should be executed by the server-side data portal.
+    /// injection.
+    /// </summary>
+    public List<IInterceptDataPortal> InterceptorProviders { get; } = new List<IInterceptDataPortal>();
+
+    /// <summary>
+    /// Gets or sets the type of the ExceptionInspector.
+    /// </summary>
+    public Type ExceptionInspectorType { get; set; }
```

Comment:
Instead of a public property of type Type, I wonder if we should use a generic method, perhaps RegisterExceptionInspector<T>(). Generic methods have the advantage that they can have constraints applied. This results in compile-time checks that the type implements the required interface, which avoids runtime exceptions caused by configuration using a type that fails to meet the required contract.

This applies to the public ExceptionInspectorType, ActivatorType and ObjectFactoryLoaderType properties.

Reply:
How would this work exactly? We don't want the calling code to create an instance, and I don't think we can restrict the type of the Type being provided?

Reply:
Doh! I see 😳